client/core/blockchain.go

```diff
@@ -1196,6 +1194,13 @@ func countTransactions(chain []*types.Block) (c int) {
 // to be part of the new canonical chain and accumulates potential missing transactions and post an
 // event about them
 func (bc *BlockChain) reorg(oldBlock, newBlock *types.Block) error {
+	if oldBlock.Number().Cmp(newBlock.Number()) == 0 && bytes.Equal(oldBlock.Hash().Bytes(), newBlock.Hash().Bytes()) {
```

Comment:
what was this doing then?

Reply:
Checks are current and received blocks the same. Will refactor a bit to make it more clear.